src/OracleService/Protocols/OracleNeoFSProtocol.cs

```diff
@@ -0,0 +1,128 @@
+        }
+
+        public void Configure()
+        {
+        }
+
+        public void Dispose()
+        {
+            privateKey.Dispose();
+        }
+
+        public async Task<(OracleResponseCode, string)> ProcessAsync(Uri uri, CancellationToken cancellation)
+        {
+            Utility.Log(nameof(OracleNeoFSProtocol), LogLevel.Debug, $"Request: {uri.AbsoluteUri}");
+
+            if (!Settings.Default.AllowPrivateHost)
+            {
+                IPHostEntry entry = await Dns.GetHostEntryAsync(uri.Host);
+                if (entry.IsInternal())
+                    return (OracleResponseCode.Forbidden, null);
+            }
+            try
+            {
+                byte[] res = await GetAsync(uri, Settings.Default.NeoFS.EndPoint, cancellation);
+                Utility.Log(nameof(OracleNeoFSProtocol), LogLevel.Debug, $"NeoFS result: {res.ToHexString()}");
+                return (OracleResponseCode.Success, Convert.ToBase64String(res));
+            }
+            catch (Exception e)
+            {
+                Utility.Log(nameof(OracleNeoFSProtocol), LogLevel.Debug, $"NeoFS result: error,{e.Message}");
+                return (OracleResponseCode.Error, null);
+            }
+        }
+
+        private async Task<byte[]> GetAsync(Uri uri, string host, CancellationToken cancellation)
+        {
+            string[] ps = uri.AbsolutePath.Split("/");
+            if (ps.Length < 2) throw new FormatException("Invalid neofs url");
+            ContainerID containerID = ContainerID.FromBase58String(ps[0]);
+            ObjectID objectID = ObjectID.FromBase58String(ps[1]);
+            Address objectAddr = new()
+            {
+                ContainerId = containerID,
+                ObjectId = objectID
+            };
+            Client client = new(privateKey, host, 120000);
+            var tokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellation);
+            tokenSource.CancelAfter(Settings.Default.NeoFS.Timeout);
+            if (ps.Length == 2)
+                return await GetPayloadAsync(client, objectAddr, tokenSource.Token);
+            return ps[2] switch
+            {
+                "range" => await GetRangeAsync(client, objectAddr, ps.Skip(3).ToArray(), tokenSource.Token),
+                "header" => GetHeader(client, objectAddr, tokenSource.Token),
+                "hash" => GetHash(client, objectAddr, ps.Skip(3).ToArray(), tokenSource.Token),
+                _ => throw new Exception("invalid command")
+            };
+        }
+
+        private static async Task<byte[]> GetPayloadAsync(Client client, Address addr, CancellationToken cancellation)
+        {
+            Object obj = await client.GetObject(cancellation, new GetObjectParams() { Address = addr }, new CallOptions { Ttl = 2 });
+            return obj.Payload.ToByteArray();
+        }
+
+        private static Task<byte[]> GetRangeAsync(Client client, Address addr, string[] ps, CancellationToken cancellation)
```

Comment:
```suggestion
        private static async Task<byte[]> GetRangeAsync(Client client, Address addr, string[] ps, CancellationToken cancellation)
```

Reply:
We don't need it be async here.

Reply:
Regardless of whether it is an `async` method, our purpose is to return a `Task`. Now since it can return a `Task` directly, it is not necessary as an `async` method. Because `async` methods have additional overhead (although very low).